src/core/action.ts

```diff
@@ -8,6 +8,11 @@ export type ISerializedActionCall = {
 }
 
 export type IRawActionCall = {
+    /**
+     * AsyncId indicates whether this action is part of an `async` based action. Id is zero if it isn't.
+     */
+    asyncId: number
+    asyncMode: "none" | "start" | "yield" | "end"
```

Comment:
What about use mode: invoke | start | yield | end and transaction: number (or random string)? I'd always generate an id even for plain actions, and on standard actions use invoke or return, whilst in async have a series of start, yield, throw, return?